### PR TITLE
Mesh3d colorscale settings

### PR DESCRIPTION
**Pull request by etpinard**
resolves https://github.com/plotly/plotly.js/issues/1710

requires:

- https://github.com/gl-vis/gl-mesh3d/pull/10 set for `gl-mesh3d@1.3.0`
- https://github.com/gl-vis/gl-error3d/pull/3 set for `gl-error3d@1.0.6`

### CODE REVIEW

File: test/image/mocks/gl3d_tetrahedra.json [hunk 1]
```diff
@@ -16,6 +16,8 @@
         [1, "rgb(0, 0, 255)"]
       ],
       "intensity": [0, 0.33, 0.66, 1],
+      "cmin": -0.5,
+      "cmax": 0.5,
```
Comment (etpinard) [inline test/image/mocks/gl3d_tetrahedra.json]: Weird. That made the axis labels disappear. I might have to revert back those `gl-error3d` dependencies bump of commit https://github.com/gl-vis/gl-error3d/commit/d756284bbfdaf3dd1f4ce7e69ae8085b9b6d2a64

Comment (etpinard) [inline test/image/mocks/gl3d_tetrahedra.json]: fixed in f55c6f5dad3c302e265a246d47c2b15d563daa52 via https://github.com/gl-vis/gl-error3d/pull/3

**Comment by etpinard** [comment]
All green on CI, ready for review @alexcjohnson

File: test/jasmine/tests/mesh3d_test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        })
+        .then(function() {
+            _assert([false, 0, 10], [false, 0, 10]);
+
+            return Plotly.restyle(gd, 'cauto', true);
+        })
+        .then(function() {
+            _assert([true, 0, 3], [true, 0, 3]);
+
+            return Plotly.purge(gd);
```
Comment (alexcjohnson) [inline test/jasmine/tests/mesh3d_test.js]: why purge?

File: src/traces/mesh3d/defaults.js [hunk 3]
```diff
@@ -77,23 +76,12 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
 
     if('intensity' in traceIn) {
         coerce('intensity');
-        coerce('showscale', true);
-    }
-    else {
+        colorscaleDefaults(traceIn, traceOut, layout, coerce, {prefix: '', cLetter: 'c'});
```
Comment (alexcjohnson) [inline src/traces/mesh3d/defaults.js]: nice when you can remove code to add features and fix bugs 🎉

**Comment by alexcjohnson** [comment]
So I guess previously these colorscale ranges were erroneously taken from the `z` span of the data? Looks great! 💃

File: test/jasmine/tests/mesh3d_test.js (hunk 2 above)
Comment (etpinard) [inline test/jasmine/tests/mesh3d_test.js]: To destroy the webgl context. 

We should perhaps add that Plotly.purge statement in destroyGraphDiv?

**Comment by etpinard** [comment]
> So I guess previously these colorscale ranges were erroneously taken from the z span of the data?

Exactly. `mesh3d` was using the heatmap [colorbar](https://github.com/plotly/plotly.js/blob/master/src/traces/heatmap/colorbar.js) logic previously.

File: test/jasmine/tests/mesh3d_test.js (hunk 2 above)
Comment (alexcjohnson) [inline test/jasmine/tests/mesh3d_test.js]: ah interesting... yeah, would be good to put that in `destroyGraphDiv` - but it's frustrating that deleting `gd` from the document isn't sufficient to 🔪 the context - I wonder if we can do something to [watch for removing](https://stackoverflow.com/questions/20156453/how-to-detect-element-being-added-removed-from-dom-element) `gd` and purge automatically? If you think that's not a good idea, we should at least make sure this is well documented somewhere prominent.

**Pull request merged**